### PR TITLE
fix(kuma-cp): fix address check to not be loopback ipv4 and ipv6

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review
There was a bug that we didn't correctly check for WorkloadIP. Now we are setting `upstreamBind` only when the transparent proxy and the `WorkloadIP` is different than loopback v4 and v6.

- [X] [Link to relevant issue][1] as well as docs and UI issues -- fix: https://github.com/kumahq/kuma/issues/8483
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by nicoche** [comment]
Hey! Thanks for looking into this. Heads up, the same condition exists in `pkg/hds/tracker/healthcheck_generator.go`

File: pkg/xds/generator/inbound_proxy_generator.go [hunk 1]
```diff
@@ -42,7 +42,8 @@ func (g InboundProxyGenerator) Generate(ctx context.Context, _ *core_xds.Resourc
 			Configure(envoy_clusters.ProvidedEndpointCluster(false, core_xds.Endpoint{Target: endpoint.WorkloadIP, Port: endpoint.WorkloadPort})).
 			Configure(envoy_clusters.Timeout(defaults_mesh.DefaultInboundTimeout(), protocol))
 		// localhost traffic is routed dirrectly to the application, in case of other interface we are going to set source address to
-		if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP != core_mesh.IPv4Loopback.String() || endpoint.WorkloadIP != core_mesh.IPv6Loopback.String()) {
+		// 127.0.0.6 to avoid redirections and thanks to first iptables rule just return fast
+		if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP != core_mesh.IPv4Loopback.String() && endpoint.WorkloadIP != core_mesh.IPv6Loopback.String()) {
```
Comment (lahabana) [inline pkg/xds/generator/inbound_proxy_generator.go:46]: @nicoche I think that's something you pointed out so would love a review :)

Comment (nicoche) [reply pkg/xds/generator/inbound_proxy_generator.go:46]: tbh I don't fully grasp how this all is supposed to work; I'm not familiar with k8s setup nor with the default iptables, so take my review with a grain of salt.

In our case, `WorkloadIP` is a `10.0. ...` address. We would validate this new condition and Envoy's configuration would have:

```
"upstream_bind_config": {
  "source_address": {
    "address": "127.0.0.6",
    "port": 0,
  },
}
```

This does not work for us as trying to initiate a connection with a loopback source address and a non-loopback destination address seems illegal. Envoy requests fail and log `invalid argument`.

This is why I was thinking that the expected condition should be `if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP == core_mesh.IPv4Loopback.String() || endpoint.WorkloadIP == core_mesh.IPv6Loopback.String()) {`

But maybe we're just doing something esoteric? 😕

Comment (jakubdyszkiewicz) [reply pkg/xds/generator/inbound_proxy_generator.go:46]: @bartsmykla to double check please

Comment (bartsmykla) [reply pkg/xds/generator/inbound_proxy_generator.go:46]: It's correct

**Pull request merged**